### DISCUSSION TITLE
When you have a stable version (aprox.) - and it's possible to upgrade after the 1st hack ?

### DISCUSSION THREAD

**Discussion by PEPER29** [bangcorrupt/hacktribe#25] [Q&A] [answered]
all it' s on title ^^

**Comment by bangcorrupt** [comment]
> stable version

This will always be a hack.

**Comment by gradientledge** [comment]
so when updating do you just reflash vanilla and do the hack again was wondering this as well.

**Comment by bangcorrupt** [comment, marked as answer]
Apply the patch to factory sampler firmware 2.02 every time.  You can update hacktribe without flashing back to factory firmware.

**Reply by gradientledge** [reply]
ok awesome, so far so good, so is this hack going to tackle the ui and sample import system? i imagine scripting for this is crazy, kinda wonder whats planned for this feature wise.  Will this eventually overhaul the file manager and file transfer to something like how an mpc does things. ie jjos? people would happily pay for this if it gets that far.

**Reply by bangcorrupt** [reply]
I haven't looked at the UI yet, there is a lot of interaction when a control is changed, making it difficult to modify.  Adding menu items might be slightly easier but I haven't tried yet.  I know how SysEx messages are handled and have basic read/write to the flash and CPU RAM.  The things I've added will be configurable via SysEx long before they are options on the device.  I may be able to do sample import via SysEx but it's slightly more complex as the data needs passing to the DSP RAM. 

I haven't used an MPC, I think the person who made JJOS had access to the source code.  I'm just levering assembly instructions into the compiled binary.  Anything labelled `enhancement` is a feature I'd like to add.  Assume nothing will change, until it does.

**Comment by gradientledge** [comment]
oh ok so kinda like how you can edit the patches on a novation circuit with hex
